src/connection/security/scram/scram-authentication.ts

```diff
@@ -73,14 +92,14 @@ export class ScramAuthentication {
    * @see https://datatracker.ietf.org/doc/html/rfc5802#section-5.1
    * @param serverFirstMessage
    */
-  private getSaltedPassword(serverFirstMessage: ServerFirstMessage): Buffer {
+  private async getSaltedPassword(serverFirstMessage: ServerFirstMessage): Promise<Buffer> {
     if (serverFirstMessage.iterations < this.mechanism.minimumIterations) {
       throw new Error(
         `Requested iterations ${serverFirstMessage.iterations} is less than the minimum ${this.mechanism.minimumIterations} for ${this.mechanism.name}`
       );
     }
 
-    return pbkdf2Sync(
+    return await promisify(pbkdf2)(
```

Comment:
save promisified function to variable and use it here, await has no meaning here